apps/aecore/test/aec_miner_tests.erl

```diff
@@ -0,0 +1,119 @@
+             %% block timestamps are rounded to seconds. Use millisecs here.
+             meck:expect(aeu_time, msecs_to_secs, fun(Ms) -> Ms end),
+             meck:expect(aec_governance, expected_block_mine_rate,
+                         fun() ->
+                                 meck:passthrough([])
+                         end),
+             TmpKeysDir = mktempd(),
+             ok = application:ensure_started(crypto),
+             {ok, _} = aec_keys:start_link(["mypassword", TmpKeysDir]),
+             {ok, _} = ?TEST_MODULE:start_link(),
+             TmpKeysDir
+     end,
+     fun(TmpKeysDir) ->
+             ok = ?TEST_MODULE:stop(),
+             ok = aec_keys:stop(),
+             ok = application:stop(crypto),
+             {ok, KeyFiles} = file:list_dir(TmpKeysDir),
+             %% Expect two filenames - private and public keys.
+             [_KF1, _KF2] = KeyFiles,
+             lists:foreach(
+               fun(F) ->
+                       AbsF = filename:absname_join(TmpKeysDir, F),
+                       {ok, _} = {file:delete(AbsF), {F, AbsF}}
+               end,
+               KeyFiles),
+             ok = file:del_dir(TmpKeysDir),
+             ?assert(meck:validate(aec_chain)),
+             ?assert(meck:validate(aec_governance)),
+             ?assert(meck:validate(aeu_time)),
+             meck:unload(aec_governance),
+             meck:unload(aeu_time),
+             meck:unload(aec_chain),
+             file:delete(TmpKeysDir)
+     end,
+     [fun(_) ->
+             [{"Start and stop",
+               fun() ->
+                       ?assertEqual(ok, ?TEST_MODULE:start_mining()),
+                       ?assertEqual(ok, ?TEST_MODULE:stop_mining()),
+                       ?assertEqual(ok, ?TEST_MODULE:start_mining()),
+                       ?assertEqual(ok, ?TEST_MODULE:stop_mining())
+               end}
+             ]
+      end,
+      fun(_) ->
+              [{"Start twice",
+               fun() ->
+                       ?assertEqual(ok, ?TEST_MODULE:start_mining()),
+                       ?assertEqual({error,already_started}, ?TEST_MODULE:start_mining()),
+                       ?assertEqual(ok, ?TEST_MODULE:stop_mining())
+               end},
+              {"Stop twice",
+               fun() ->
+                       ?assertEqual(ok, ?TEST_MODULE:start_mining()),
+                       ?assertEqual(ok, ?TEST_MODULE:stop_mining()),
+                       ?assertEqual({error,not_started}, ?TEST_MODULE:stop_mining())
+               end},
+              {"Stop in idle",
+               fun() ->
+                       ?assertEqual({error,not_started}, ?TEST_MODULE:stop_mining())
+               end},
+              {timeout, 60,
+               {"Run miner for a while",
+                fun() ->
+                        ?assertEqual(ok, ?TEST_MODULE:start_mining()),
+                        timer:sleep(20000),
```

Comment:
I understand this is 20 seconds (!!!). Could you please add TODO comment to speed up this test e.g. by making configurable number of mining attempts and making it event driven rather than time driven?